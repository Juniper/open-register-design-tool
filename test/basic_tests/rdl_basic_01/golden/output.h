//   Ordt 230719.01 autogenerated file 
//   Input: ./rdl_basic_01/test.rdl
//   Parms: ./rdl_basic_01/test.parms
//   Date: Thu Jul 20 13:46:40 EDT 2023
//

#ifndef __SIMPLE1_REGISTER_MAP__
#define __SIMPLE1_REGISTER_MAP__

#include <bits.h>

/* SIMPLE1_REGISTERS memory map */
enum SIMPLE1_REGS {
  STATS_DISABLE_CHECK_0 = 0x1000,
  STATS_DISABLE_CHECK_1 = 0x1080,
  STATS_FEATURES = 0x1100,
  STATS_LOG_ADDRESS1 = 0x1104,
  STATS_LOG_ADDRESS2 = 0x1110,
  STATS_WIDE_REG = 0x1120,
  RDR_REORDER_WINDOW = 0x4000,
  RDR_STREAM_RECONF = 0x4004,
  RDR_CP_FP_WR_BUFFER_0 = 0x61c0,
  RDR_CP_FP_WR_BUFFER_1 = 0x61c4,
  RDR_CP_FP_WR_BUFFER_2 = 0x61c8,
  RDR_CP_FP_WR_BUFFER_3 = 0x61cc,
  RDR_ROLL32_COUNTER_REG = 0x61d0,
  RDR_SAT32_COUNTER_REG = 0x61d4,
  RDR_RCNT_SAT_LOG = 0x61d8,
  EXTRA_REG = 0x61dc
};

/* STATS_DISABLE_CHECK_0 register fields */
#define STATS_DISABLE_CHECK_0_TIMER_SRAM BIT(0)
#define STATS_DISABLE_CHECK_0_CBUF_FIFO BIT(1)
#define STATS_DISABLE_CHECK_0_LUT_ST BIT(2)
#define STATS_DISABLE_CHECK_0_L0_B0 BIT(3)
#define STATS_DISABLE_CHECK_0_L0_B1 BIT(4)
#define STATS_DISABLE_CHECK_0_L0_B2 BIT(5)
#define STATS_DISABLE_CHECK_0_L0_B3 BIT(6)

/* STATS_DISABLE_CHECK_1 register fields */
#define STATS_DISABLE_CHECK_1_TIMER_SRAM BIT(0)
#define STATS_DISABLE_CHECK_1_CBUF_FIFO BIT(1)
#define STATS_DISABLE_CHECK_1_LUT_ST BIT(2)
#define STATS_DISABLE_CHECK_1_L0_B0 BIT(3)
#define STATS_DISABLE_CHECK_1_L0_B1 BIT(4)
#define STATS_DISABLE_CHECK_1_L0_B2 BIT(5)
#define STATS_DISABLE_CHECK_1_L0_B3 BIT(6)

/* STATS_FEATURES register fields */
#define STATS_FEATURES_VERSION GENMASK(31, 30)
#define STATS_FEATURES_DEFAULT_RST_FLD GENMASK(17, 16)
#define STATS_FEATURES_ALT_RST_FLD GENMASK(15, 14)
#define STATS_FEATURES_SYN_LOGGED BIT(9)
#define STATS_FEATURES_WDTH GENMASK(8, 4)
#define STATS_FEATURES_GENERATOR BIT(3)
#define STATS_FEATURES_CHECKER BIT(2)
#define STATS_FEATURES_CORRECTABLE BIT(1)
#define STATS_FEATURES_ADR_LOGGED BIT(0)

/* STATS_LOG_ADDRESS1 register fields */
#define STATS_LOG_ADDRESS1_VALUE GENMASK(9, 0)

/* STATS_LOG_ADDRESS2 register fields */
#define STATS_LOG_ADDRESS2_VALUE GENMASK(9, 0)

/* STATS_WIDE_REG register fields */
#define STATS_WIDE_REG_VALUE GENMASK(95, 0)

/* RDR_REORDER_WINDOW register fields */
#define RDR_REORDER_WINDOW_WS_ENABLE BIT(31)
#define RDR_REORDER_WINDOW_SET_BACK GENMASK(26, 16)
#define RDR_REORDER_WINDOW_SIZE GENMASK(10, 0)

/* RDR_STREAM_RECONF register fields */
#define RDR_STREAM_RECONF_BUSY BIT(16)
#define RDR_STREAM_RECONF_STREAM GENMASK(9, 0)

/* RDR_CP_FP_WR_BUFFER_0 register fields */
#define RDR_CP_FP_WR_BUFFER_0_ADDR0 GENMASK(26, 16)
#define RDR_CP_FP_WR_BUFFER_0_ADDR1 GENMASK(10, 0)

/* RDR_CP_FP_WR_BUFFER_1 register fields */
#define RDR_CP_FP_WR_BUFFER_1_ADDR0 GENMASK(26, 16)
#define RDR_CP_FP_WR_BUFFER_1_ADDR1 GENMASK(10, 0)

/* RDR_CP_FP_WR_BUFFER_2 register fields */
#define RDR_CP_FP_WR_BUFFER_2_ADDR0 GENMASK(26, 16)
#define RDR_CP_FP_WR_BUFFER_2_ADDR1 GENMASK(10, 0)

/* RDR_CP_FP_WR_BUFFER_3 register fields */
#define RDR_CP_FP_WR_BUFFER_3_ADDR0 GENMASK(26, 16)
#define RDR_CP_FP_WR_BUFFER_3_ADDR1 GENMASK(10, 0)

/* RDR_ROLL32_COUNTER_REG register fields */
#define RDR_ROLL32_COUNTER_REG_COUNT GENMASK(3, 0)

/* RDR_SAT32_COUNTER_REG register fields */
#define RDR_SAT32_COUNTER_REG_COUNT GENMASK(3, 0)

/* RDR_RCNT_SAT_LOG register fields */
#define RDR_RCNT_SAT_LOG_EN BIT(31)
#define RDR_RCNT_SAT_LOG_SUBCH GENMASK(30, 26)
#define RDR_RCNT_SAT_LOG_STR GENMASK(25, 16)
#define RDR_RCNT_SAT_LOG_COUNT GENMASK(15, 0)

/* EXTRA_REG (extra_reg name) register fields */
#define EXTRA_REG_VALUE GENMASK(31, 0)

#endif
