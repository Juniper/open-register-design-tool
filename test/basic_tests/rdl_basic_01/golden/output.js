/*   Ordt 190725.01 autogenerated file 
 *   Input: ./rdl_basic_01/test.rdl
 *   Parms: ./rdl_basic_01/test.parms
 *   Date: Sun Nov 17 17:01:46 EST 2019
 */

register_set simple1 "Registers for simple1" {

  address = 0x0;
  register_width = 32;
  register_set stats "stats register_set" {
    address = 0x1000;
    
    register disable_check "disable_check register" {
      address = 0x0;
      repeat = 2;
      nop[25];
      integer l0_b3[1] "l0_b3 field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer l0_b2[1] "l0_b2 field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer l0_b1[1] "l0_b1 field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer l0_b0[1] "l0_b0 field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer lut_st[1] "lut_st field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer cbuf_fifo[1] "cbuf_fifo field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer timer_sram[1] "timer_sram field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register features "features register" {
      address = 0x100;
      integer version[2] "version field" param {
        access_mode = READ_ONLY;
        reset = 0x2;
      };
      nop[12];
      integer default_rst_fld[2] "default_rst_fld field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer alt_rst_fld[2] "alt_rst_fld field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      nop[4];
      integer syn_logged[1] "syn_logged field" param {
        access_mode = READ_ONLY;
        reset = 0x0;
      };
      integer wdth[5] "wdth field" param {
        access_mode = READ_ONLY;
        reset = 0x7;
      };
      integer generator[1] "generator field" param {
        access_mode = READ_ONLY;
        reset = 0x1;
      };
      integer checker[1] "checker field" param {
        access_mode = READ_ONLY;
        reset = 0x1;
      };
      integer correctable[1] "correctable field" param {
        access_mode = READ_ONLY;
        reset = 0x0;
      };
      integer adr_logged[1] "adr_logged field" param {
        access_mode = READ_ONLY;
        reset = 0x1;
      };
    };
    
    register log_address1 "log_address1 register" {
      address = 0x104;
      category = { STATE };
      description = "{
        newdesc bla bla
      }";
      nop[22];
      integer value[10] "value field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register log_address2 "log_address2 register" {
      address = 0x110;
      category = { STATE };
      description = "{
        newdesc 2 bla bla
      }";
      nop[22];
      integer value[10] "value field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register wide_reg "wide_reg register" {
      address = 0x120;
      register_width = 128;
      nop[32];
      integer value[96] "value field" param {
        access_mode = READ_WRITE;
        reset = 0xa5a5a5a5a5a5a5a5a5a5a5a5;
      };
    };
    
    register_set_size = 0x130;
  };
  
  register_set rdr "rdr register_set" {
    address = 0x4000;
    
    register reorder_window "reorder_window register" {
      address = 0x0;
      category = { STATE };
      integer ws_enable[1] "ws_enable field" param {
        description = "{
          SW Configuration field
        }";
        access_mode = READ_WRITE;
        reset = 0x1;
      };
      nop[4];
      integer set_back[11] "set_back field" param {
        description = "{
          SW Configuration field
        }";
        access_mode = READ_WRITE;
        reset = 0x10;
      };
      nop[5];
      integer size[11] "size field" param {
        description = "{
          SW Configuration field
        }";
        access_mode = READ_WRITE;
        reset = 0x80;
      };
    };
    
    register stream_reconf "stream_reconf register" {
      address = 0x4;
      nop[15];
      integer busy[1] "busy field" param {
        description = "{
          HW Status field
        }";
        access_mode = READ_ONLY;
        reset = 0x0;
      };
      nop[6];
      integer stream[10] "stream field" param {
        description = "{
          SW Configuration field
        }";
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register_set cp_fp_wr "cp_fp_wr register_set" {
      address = 0x21c0;
      
      register buffer "buffer register" {
        address = 0x0;
        access_mode = READ_ONLY;
        category = { STATE };
        repeat = 4;
        nop[5];
        integer addr0[11] "addr0 field" param {
          description = "{
            HW Status field
          }";
          access_mode = READ_ONLY;
          reset = unknown;
        };
        nop[5];
        integer addr1[11] "addr1 field" param {
          description = "{
            HW Status field
          }";
          access_mode = READ_ONLY;
          reset = unknown;
        };
      };
      
      register_set_size = 0x10;
    };
    
    register roll32_counter_reg "roll32_counter_reg register" {
      address = 0x21d0;
      nop[28];
      integer count[4] "count field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register sat32_counter_reg "sat32_counter_reg register" {
      address = 0x21d4;
      attributes = JS_ATTRIB_TEST_ACCESS_ONLY;
      nop[28];
      integer count[4] "count field" param {
        access_mode = READ_TO_CLEAR;
        reset = unknown;
      };
    };
    
    register rcnt_sat_log "rcnt_sat_log register" {
      address = 0x21d8;
      integer en[1] "en field" param {
        description = "{
          SW Configuration field
        }";
        access_mode = READ_WRITE;
        reset = 0x1;
      };
      integer subch[5] "subch field" param {
        access_mode = READ_TO_CLEAR;
        reset = 0x0;
      };
      integer str[10] "str field" param {
        access_mode = READ_TO_CLEAR;
        sub_category = { INFO };
        reset = 0x0;
      };
      integer count[16] "count field" param {
        access_mode = READ_TO_CLEAR;
        reset = 0x0;
      };
    };
    
    register_set_size = 0x21dc;
  };
  
  register extra_reg "extra_reg name" {
    address = 0x61dc;
    category = { DYNAMIC_CONFIG };
    description = "{
      extra_reg description bla bla bla
                more bla
    }";
    integer value[32] "value field" param {
      description = "{
        SW Configuration field
      }";
      access_mode = READ_WRITE;
      reset = unknown;
    };
  };
  
  register_set_size = 0x61e0;
};

