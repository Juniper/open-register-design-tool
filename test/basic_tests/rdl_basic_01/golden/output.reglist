//   Ordt 190725.01 autogenerated file 
//   Input: ./rdl_basic_01/test.rdl
//   Parms: ./rdl_basic_01/test.parms
//   Date: Sun Nov 17 17:02:48 EST 2019
//

0x1000           size: 32   fields: 7    bits: 7         stats.disable_check_0
0x1080           size: 32   fields: 7    bits: 7         stats.disable_check_1
0x1100           size: 32   fields: 9    bits: 16        stats.features
0x1104           size: 32   fields: 1    bits: 10        stats.log_address1
0x1110           size: 32   fields: 1    bits: 10        stats.log_address2
0x1120           size: 128  fields: 1    bits: 96        stats.wide_reg
0x4000           size: 32   fields: 3    bits: 23        rdr.reorder_window
0x4004           size: 32   fields: 2    bits: 11        rdr.stream_reconf
0x61c0           size: 32   fields: 2    bits: 22        rdr.cp_fp_wr.buffer_0
0x61c4           size: 32   fields: 2    bits: 22        rdr.cp_fp_wr.buffer_1
0x61c8           size: 32   fields: 2    bits: 22        rdr.cp_fp_wr.buffer_2
0x61cc           size: 32   fields: 2    bits: 22        rdr.cp_fp_wr.buffer_3
0x61d0           size: 32   fields: 1    bits: 4         rdr.roll32_counter_reg
0x61d4           size: 32   fields: 1    bits: 4         rdr.sat32_counter_reg
0x61d8           size: 32   fields: 4    bits: 32        rdr.rcnt_sat_log
0x61dc           size: 32   fields: 1    bits: 32        extra_reg
