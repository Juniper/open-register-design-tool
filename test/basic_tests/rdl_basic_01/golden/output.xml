<?xml version="1.0" encoding="UTF-8"?>
<!--   Ordt 191120.01 autogenerated file 
       Input: ./rdl_basic_01/test.rdl
       Parms: ./rdl_basic_01/test.parms
       Date: Wed Nov 20 22:19:43 EST 2019
 -->

<map version="191120.01">
  <id>simple1</id>
  <baseaddr>0</baseaddr>
  <shorttext>simple1 registers</shorttext>
  <regset>
    <id>stats</id>
    <shorttext>stats registers</shorttext>
    <baseaddr>0x1000</baseaddr>
    <reg>
      <id>disable_check</id>
      <parentpath>stats</parentpath>
      <shorttext>disable_check register</shorttext>
      <baseaddr>0x1000</baseaddr>
      <width>32</width>
      <reps>2</reps>
      <stride>0x80</stride>
      <field>
        <id>timer_sram</id>
        <shorttext>timer_sram field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>cbuf_fifo</id>
        <shorttext>cbuf_fifo field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>1</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>lut_st</id>
        <shorttext>lut_st field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>2</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>l0_b0</id>
        <shorttext>l0_b0 field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>3</lowidx>
        <width>1</width>
        <hwmod></hwmod>
        <intr>
          <type>LEVEL</type>
          <stickytype>STICKYBIT</stickytype>
        </intr>
      </field>
      <field>
        <id>l0_b1</id>
        <shorttext>l0_b1 field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>4</lowidx>
        <width>1</width>
        <hwmod></hwmod>
        <intr>
          <type>LEVEL</type>
          <stickytype>STICKYBIT</stickytype>
        </intr>
      </field>
      <field>
        <id>l0_b2</id>
        <shorttext>l0_b2 field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>5</lowidx>
        <width>1</width>
        <hwmod></hwmod>
        <intr>
          <type>LEVEL</type>
          <stickytype>STICKYBIT</stickytype>
          <halt></halt>
        </intr>
      </field>
      <field>
        <id>l0_b3</id>
        <shorttext>l0_b3 field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>6</lowidx>
        <width>1</width>
      </field>
      <access>RW</access>
    </reg>
    <reg>
      <id>features</id>
      <parentpath>stats</parentpath>
      <shorttext>features register</shorttext>
      <baseaddr>0x1100</baseaddr>
      <width>32</width>
      <field>
        <id>version</id>
        <shorttext>version field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
        </hwinfo>
        <reset>0x2</reset>
        <lowidx>30</lowidx>
        <width>2</width>
      </field>
      <field>
        <id>default_rst_fld</id>
        <shorttext>default_rst_fld field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>16</lowidx>
        <width>2</width>
      </field>
      <field>
        <id>alt_rst_fld</id>
        <shorttext>alt_rst_fld field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>14</lowidx>
        <width>2</width>
      </field>
      <field>
        <id>syn_logged</id>
        <shorttext>syn_logged field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>9</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>wdth</id>
        <shorttext>wdth field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
          <ored></ored>
        </hwinfo>
        <reset>0x7</reset>
        <lowidx>4</lowidx>
        <width>5</width>
      </field>
      <field>
        <id>generator</id>
        <shorttext>generator field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
          <anded></anded>
        </hwinfo>
        <reset>0x1</reset>
        <lowidx>3</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>checker</id>
        <shorttext>checker field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
          <xored></xored>
        </hwinfo>
        <reset>0x1</reset>
        <lowidx>2</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>correctable</id>
        <shorttext>correctable field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>1</lowidx>
        <width>1</width>
      </field>
      <field>
        <id>adr_logged</id>
        <shorttext>adr_logged field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
        </hwinfo>
        <reset>0x1</reset>
        <lowidx>0</lowidx>
        <width>1</width>
      </field>
      <access>RW</access>
    </reg>
    <reg>
      <id>log_address1</id>
      <parentpath>stats</parentpath>
      <shorttext>log_address1 register</shorttext>
      <baseaddr>0x1104</baseaddr>
      <width>32</width>
      <longtext><![CDATA[newdesc bla bla]]></longtext>
      <field>
        <id>value</id>
        <shorttext>value field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>w</hwaccess>
          <we></we>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>10</width>
        <hwmod></hwmod>
      </field>
      <catcode>32</catcode>
      <access>RW</access>
    </reg>
    <reg>
      <id>log_address2</id>
      <parentpath>stats</parentpath>
      <shorttext>log_address2 register</shorttext>
      <baseaddr>0x1110</baseaddr>
      <width>32</width>
      <longtext><![CDATA[newdesc 2 bla bla]]></longtext>
      <field>
        <id>value</id>
        <shorttext>value field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>w</hwaccess>
          <we></we>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>10</width>
        <hwmod></hwmod>
      </field>
      <catcode>32</catcode>
      <access>RW</access>
    </reg>
    <reg>
      <id>wide_reg</id>
      <parentpath>stats</parentpath>
      <shorttext>wide_reg register</shorttext>
      <baseaddr>0x1120</baseaddr>
      <width>128</width>
      <field>
        <id>value</id>
        <shorttext>value field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>rw</hwaccess>
          <we></we>
        </hwinfo>
        <reset>0xa5a5a5a5a5a5a5a5a5a5a5a5</reset>
        <lowidx>0</lowidx>
        <width>96</width>
        <hwmod></hwmod>
      </field>
      <access>RW</access>
    </reg>
    <highaddr>0x112f</highaddr>
  </regset>
  <regset>
    <id>rdr</id>
    <shorttext>rdr registers</shorttext>
    <baseaddr>0x4000</baseaddr>
    <reg>
      <id>reorder_window</id>
      <parentpath>rdr</parentpath>
      <shorttext>reorder_window register</shorttext>
      <baseaddr>0x4000</baseaddr>
      <width>32</width>
      <field>
        <id>ws_enable</id>
        <shorttext>ws_enable field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x1</reset>
        <lowidx>31</lowidx>
        <width>1</width>
        <longtext><![CDATA[SW Configuration field]]></longtext>
      </field>
      <field>
        <id>set_back</id>
        <shorttext>set_back field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x10</reset>
        <lowidx>16</lowidx>
        <width>11</width>
        <longtext><![CDATA[SW Configuration field]]></longtext>
      </field>
      <field>
        <id>size</id>
        <shorttext>size field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x80</reset>
        <lowidx>0</lowidx>
        <width>11</width>
        <longtext><![CDATA[SW Configuration field]]></longtext>
      </field>
      <catcode>32</catcode>
      <access>RW</access>
    </reg>
    <reg>
      <id>stream_reconf</id>
      <parentpath>rdr</parentpath>
      <shorttext>stream_reconf register</shorttext>
      <baseaddr>0x4004</baseaddr>
      <width>32</width>
      <field>
        <id>busy</id>
        <shorttext>busy field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>w</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>16</lowidx>
        <width>1</width>
        <hwmod></hwmod>
        <longtext><![CDATA[HW Status field]]></longtext>
      </field>
      <field>
        <id>stream</id>
        <shorttext>stream field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>10</width>
        <longtext><![CDATA[SW Configuration field]]></longtext>
      </field>
      <access>RW</access>
    </reg>
    <regset>
      <id>cp_fp_wr</id>
      <parentpath>rdr</parentpath>
      <shorttext>cp_fp_wr registers</shorttext>
      <baseaddr>0x61c0</baseaddr>
      <extroot>1</extroot>
      <reg>
        <id>buffer</id>
        <parentpath>rdr.cp_fp_wr</parentpath>
        <shorttext>buffer register</shorttext>
        <baseaddr>0x61c0</baseaddr>
        <width>32</width>
        <reps>4</reps>
        <stride>0x4</stride>
        <field>
          <id>addr0</id>
          <shorttext>addr0 field</shorttext>
          <access>RO</access>
          <hwinfo>
            <hwaccess>w</hwaccess>
          </hwinfo>
          <lowidx>16</lowidx>
          <width>11</width>
          <hwmod></hwmod>
          <longtext><![CDATA[HW Status field]]></longtext>
        </field>
        <field>
          <id>addr1</id>
          <shorttext>addr1 field</shorttext>
          <access>RO</access>
          <hwinfo>
            <hwaccess>w</hwaccess>
          </hwinfo>
          <lowidx>0</lowidx>
          <width>11</width>
          <hwmod></hwmod>
          <longtext><![CDATA[HW Status field]]></longtext>
        </field>
        <catcode>32</catcode>
        <access>RO</access>
      </reg>
      <highaddr>0x61cf</highaddr>
    </regset>
    <reg>
      <id>roll32_counter_reg</id>
      <parentpath>rdr</parentpath>
      <shorttext>roll32_counter_reg register</shorttext>
      <baseaddr>0x61d0</baseaddr>
      <width>32</width>
      <field>
        <id>count</id>
        <shorttext>count field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>rw</hwaccess>
          <we></we>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>4</width>
        <hwmod></hwmod>
        <counter>
          <incr>
            <overflow></overflow>
            <incrvalue>0x1</incrvalue>
          </incr>
        </counter>
      </field>
      <access>RW</access>
    </reg>
    <reg>
      <id>sat32_counter_reg</id>
      <parentpath>rdr</parentpath>
      <shorttext>sat32_counter_reg register</shorttext>
      <baseaddr>0x61d4</baseaddr>
      <width>32</width>
      <field>
        <id>count</id>
        <shorttext>count field</shorttext>
        <access>WRC</access>
        <hwinfo>
          <hwaccess>rw</hwaccess>
          <we></we>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>4</width>
        <hwmod></hwmod>
        <dontcompare></dontcompare>
        <counter>
          <incr>
            <saturate>no_output</saturate>
            <incrvalue>0x1</incrvalue>
            <satvalue>0xf</satvalue>
            <threshold>0x8</threshold>
          </incr>
        </counter>
      </field>
      <access>WRC</access>
    </reg>
    <reg>
      <id>rcnt_sat_log</id>
      <parentpath>rdr</parentpath>
      <shorttext>rcnt_sat_log register</shorttext>
      <baseaddr>0x61d8</baseaddr>
      <width>32</width>
      <field>
        <id>en</id>
        <shorttext>en field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x1</reset>
        <lowidx>31</lowidx>
        <width>1</width>
        <longtext><![CDATA[SW Configuration field]]></longtext>
      </field>
      <field>
        <id>subch</id>
        <shorttext>subch field</shorttext>
        <access>WRC</access>
        <hwinfo>
          <hwaccess>w</hwaccess>
          <we></we>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>26</lowidx>
        <width>5</width>
        <hwmod></hwmod>
      </field>
      <field>
        <id>str</id>
        <shorttext>str field</shorttext>
        <access>WRC</access>
        <hwinfo>
          <hwaccess>w</hwaccess>
          <we></we>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>16</lowidx>
        <width>10</width>
        <hwmod></hwmod>
        <subcatcode>1</subcatcode>
      </field>
      <field>
        <id>count</id>
        <shorttext>count field</shorttext>
        <access>WRC</access>
        <hwinfo>
          <hwaccess>rw</hwaccess>
          <we></we>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>16</width>
        <hwmod></hwmod>
        <counter>
          <incr>
            <saturate>no_output</saturate>
            <incrvalue>0x1</incrvalue>
            <satvalue>0xffff</satvalue>
          </incr>
        </counter>
      </field>
      <access>RW</access>
    </reg>
    <highaddr>0x61db</highaddr>
  </regset>
  <reg>
    <id>extra_reg</id>
    <shorttext>extra_reg name</shorttext>
    <baseaddr>0x61dc</baseaddr>
    <width>32</width>
    <longtext><![CDATA[extra_reg description bla bla bla
                more bla]]></longtext>
    <field>
      <id>value</id>
      <shorttext>value field</shorttext>
      <access>RW</access>
      <hwinfo>
        <hwaccess>r</hwaccess>
      </hwinfo>
      <lowidx>0</lowidx>
      <width>32</width>
      <longtext><![CDATA[SW Configuration field]]></longtext>
    </field>
    <catcode>2</catcode>
    <access>RW</access>
  </reg>
</map>
